docs/architecture/adr-064-abci-2.0.md

```diff
@@ -353,13 +353,17 @@ we can come up with new types and names altogether.
 func (app *BaseApp) FinalizeBlock(req abci.RequestFinalizeBlock) (*abci.ResponseFinalizeBlock, error) {
 	ctx := ...
 
-	if app.preFinalizeBlockHook != nil {
-		if err := app.preFinalizeBlockHook(ctx, req); err != nil {
+	if app.preBlocker != nil {
+		ctx := app.finalizeBlockState.ctx
+		rsp, err := app.preBlocker(ctx, req)
+		if err != nil {
 			return nil, err
 		}
+		if rsp.ConsensusParamsChanged {
+			app.finalizeBlockState.ctx = ctx.WithConsensusParams(app.GetConsensusParams(ctx))
+		}
```

Comment:
q: Might be a silly one but... Why do we need this, couldn't we just always do it? Or are there cases in which we want to update params but not make those changes available until the next block?

Reply:
we only need reload when consensus params get modified in preBlocker

Reply:
Yeah, but what difference would it make in practice to always reload the consensus params? That way we could discard `ResponsePreBlock`

Reply:
Ah, I realize I'm late to the party. This was in the previous PR, and I only thought of this now 🤦

Reply:
but this is a rare case and should in reality only happen in upgrades